src/App.tsx

```diff
@@ -157,239 +286,541 @@ function App() {
+
+                            // inject the newly selected address
+                            exampleParams = insertSigningAddress(
+                              currentMethod,
+                              exampleParams,
+                              newAddress,
+                            );
+
+                            const updatedRequest = {
+                              method: 'wallet_invokeMethod',
+                              params: {
+                                scope,
+                                request: exampleParams,
+                              },
+                            };
+
+                            setInvokeMethodRequests((prev) => ({
+                              ...prev,
+                              [scope]: JSON.stringify(updatedRequest, null, 2),
+                            }));
+                          }
+                        }
+                      }}
+                    >
+                      <option value="">Select an account</option>
+                      {(details.accounts ?? []).map((account: string) => {
+                        const address = account.split(':').pop();
+                        return (
+                          <option key={account} value={account}>
+                            {address}
+                          </option>
+                        );
+                      })}
+                    </select>
+
+                    <select
+                      value={selectedMethods[scope] ?? ''}
+                      onChange={(evt) => handleMethodSelect(evt, scope)}
+                    >
+                      <option value="">Select a method</option>
+                      {details.methods.map((method: string) => (
+                        <option key={method} value={method}>
+                          {method}
+                        </option>
+                      ))}
+                    </select>
+
+                    <details className="collapsible-section">
+                      <summary>Invoke Method Request</summary>
+                      <div className="collapsible-content">
+                        <textarea
+                          value={invokeMethodRequests[scope] ?? ''}
+                          onChange={(evt) =>
+                            setInvokeMethodRequests((prev) => ({
+                              ...prev,
+                              [scope]: evt.target.value,
+                            }))
+                          }
+                          rows={5}
+                          cols={50}
+                        />
+                      </div>
+                    </details>
+
+                    <button
+                      id={`invoke-method-${scope}-btn`}
+                      onClick={async () => {
+                        if (selectedMethods[scope]) {
+                          await handleInvokeMethod(
+                            scope,
+                            selectedMethods[scope] ?? '',
```

Comment:
is the nullish coalesce needed given the guard?

Reply:
removed here: done here: https://github.com/MetaMask/test-dapp-multichain/pull/15/commits/37b80aad7b01e6a0fdcb4f649d3dac278f2c9334